core/store/fs_directory.cpp

```diff
@@ -427,9 +467,15 @@ fs_index_input::file_handle::ptr pooled_fs_index_input::reopen(
   const file_handle& src) const {
   // reserve a new handle from the pool
   std::shared_ptr<fs_index_input::file_handle> handle{
-    const_cast<pooled_fs_index_input*>(this)->fd_pool_->emplace()};
-
+    const_cast<pooled_fs_index_input*>(this)->fd_pool_->emplace(
+      src.resource_manager)};
+  size_t descriptors{0};
+  irs::Finally cleanup = [&]() noexcept {
+    src.resource_manager.file_descriptors->DecreaseChecked(descriptors);
+  };
   if (!handle->handle) {
+    src.resource_manager.file_descriptors->Increase(descriptors);
+    descriptors = 1;
```

Comment:
You make increase on 0!
```suggestion
    descriptors = 1;
    src.resource_manager.file_descriptors->Increase(descriptors);
```
I also not sure that you need to decrease in case 
```
"Failed to obtain current position of input file,
```

Reply:
Yes, decrease should be cancelled as soon as handle is stored. From that point file_handle wrapper is responsible for Decrease